src/docfx/publish/PublishUrlMap.cs

```diff
@@ -117,12 +117,17 @@ private PublishUrlMapItem ResolveOutputPathConflicts(IGrouping<string, PublishUr
                 return conflicts.First();
             }
 
-            _errors.Add(Errors.UrlPath.OutputPathConflict(conflicts.First().OutputPath, conflicts.Select(x => x.SourcePath)));
-
             // redirection file is preferred than source file
+            var redirection = conflicts.SingleOrDefault(x => x.SourcePath.Origin == FileOrigin.Redirection);
```

Comment:
`SingleOrDefault` throws when there are more then 1 items match the condition (which can crash the process).
```suggestion
            var redirection = conflicts.FirstOrDefault(x => x.SourcePath.Origin == FileOrigin.Redirection);
```

Reply:
We should not have multiple redirection with same file, it will end with `redirection-conflict`?